File: backend/pokerogue/src/main/java/com/pokerogue/helper/ability2/dto/AbilityDetailResponse.java
package com.pokerogue.helper.ability2.dto;

import com.pokerogue.helper.ability2.data.Ability;
import java.util.List;

public record AbilityDetailResponse(
        String koName,
        String description,
        List<AbilityPokemonResponse> pokemons
) {

    public static AbilityDetailResponse of(Ability ability, List<AbilityPokemonResponse> pokemons) {
        return new AbilityDetailResponse(ability.getName(), ability.getDescription(), pokemons);
    }
}

File: android/remote/src/main/java/poke/rogue/helper/remote/dto/response/ability/AbilityDetailResponse.kt
package poke.rogue.helper.remote.dto.response.ability

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import poke.rogue.helper.remote.dto.response.pokemon.PokemonResponse

@Serializable
data class AbilityDetailResponse(
    @SerialName("koName")
    val title: String,
    val description: String,
    val pokemons: List<PokemonResponse>,
)
